### PR TITLE
David Caboni : string calculator challenge

### PR DESCRIPTION
**Pull request by davconid**
Hello the serenityBDD masters :)

Here is my solution on this challenge.
I did it entirely alone, without watching at all the solution, so i'm pretty sure you will have a lot of remarks. 
I'm all ears.

Now I'm going to see the solution.

Thx

### CODE REVIEW

File: src/main/java/com/serenitydojo/calculator/Calculator.java [hunk 1]
```diff
@@ -1,7 +1,90 @@
 package com.serenitydojo.calculator;
 
+import static org.apache.commons.lang3.StringUtils.isNumeric;
+
 public class Calculator {
+    private String operator = "";
+    private int evaluation = 0;
```
Comment (wakaleo) [inline src/main/java/com/serenitydojo/calculator/Calculator.java:7]: Minor point - making the evaluation variable a field means that the class will not be thread-safe - if two threads called the same object in parallel, the calculation would be wrong. It's not really relevant for this scenario, but worth knowing.

File: src/test/java/com/serenitydojo/calculator/WhenDoingMaths.java [hunk 2]
```diff
@@ -13,38 +13,38 @@ public void shouldReturnZeroForAnEmptyString() {
...
+
+    @Test
+    public void shouldAddAndSubtract() {
+        assertThat(calculator.evaluate("10 + 5 - 6")).isEqualTo(9);
+    }
+
+    @Test
+    public void shouldMultiplyNumbers() {
+        assertThat(calculator.evaluate("10 * 5")).isEqualTo(50);
+    }
```
Comment (wakaleo) [inline src/test/java/com/serenitydojo/calculator/WhenDoingMaths.java:49]: This is all great stuff - nice clear assertions.

File: src/main/java/com/serenitydojo/calculator/Calculator.java [hunk 3]
```diff
@@ -1,7 +1,90 @@
...
+        if (isNumeric(element)) {
+            int[] intFromString = gettingIntFromString(element);
+            this.evaluation = doingTheOperation(intFromString[0], intFromString[1]);
+        } else {
+            this.operator = element;
+        }
+    }
+
+
+    private int[] gettingIntFromString(String element) {
```
Comment (wakaleo) [inline src/main/java/com/serenitydojo/calculator/Calculator.java:65]: This method is a bit confusing. Methods tend to be easier to understand if they have only use input parameters, and avoid using fields, as it's hard to know where the field values (e.g. `evaluation` in this case) are doing.

**Review by wakaleo — COMMENTED** [review]
Just some general comments on the use of the member variable, which seems to make the code a bit more complicated than it could be.

File: src/test/java/com/serenitydojo/calculator/WhenDoingMaths.java (hunk 2 above)
Comment (davconid) [reply src/test/java/com/serenitydojo/calculator/WhenDoingMaths.java:49]: Thx a lot John, but I can't take the compliment : the assertions where part of the initial state of the project, I just uncommented them

File: src/main/java/com/serenitydojo/calculator/Calculator.java (hunk 3 above)
Comment (davconid) [reply src/main/java/com/serenitydojo/calculator/Calculator.java:65]: Ok, I understand. Will try to avoid field in the future. Thx.

**Pull request closed without merging**